DIPs/DIP1NNN-DK.md

```diff
@@ -0,0 +1,696 @@
+
+While writing to a `@system` variable is always unsafe, reading from one is only dangerous when it could yield an unsafe value.
+
+```D
+struct Handle {
+    @system int handle;
+}
+
+// struct with @system field is an unsafe type
+@safe   Handle safeHandle = Handle(1);
+@system Handle systemHandle = Handle(-1);
+
+// pointers are an unsafe type
+@safe   immutable int* safePtr   = null;
+@system immutable int* systemPtr = cast(int*) 0x8035FDF0;
+
+// integers are a safe type
+@safe   int safeInt   = 20;
+@system int systemInt = 20;
+
+void main() @safe {
+    Handle h0 = safeHandle;        // allowed, @safe variable
+    Handle h1 = systemHandle;      // error, reading @system var of unsafe type
+    immutable int* p0 = safePtr;   // allowed, @safe variable
+    immutable int* p1 = systemPtr; // error, reading @system var of unsafe type
+    int i0 = safeInt;              // allowed
+    int i1 = systemInt;            // allowed, not an unsafe type
+}
+```
+
+**(3) Variables and fields without annotation are `@safe` unless their initial value is not `@safe`**
+
+The rules regarding variables and fields are as follows:
+- An initialization expression `x` is `@system` when the function `(() => x)` is inferred as `@system`.
+- When marked `@system`, the result is always `@system` regardless of the type.
+- When marked `@trusted`, the initialization expression `x` is treated as `(() @trusted => x)`.
+- When marked `@safe`, the initialization expression must be `@safe`.
+- In the absence of an annotation, the result is `@system` only if the type is unsafe and the initialization expression is `@system`.
+
+```D
+int* getPtr() @system {return cast(int*) 0x8035FDF0;}
+int  getVal() @system {return -1;}
+
+extern int* x0;                   // @safe by default
+int* x1 = x0;                     // @safe, (() => x0) is @safe
+int* x2 = cast(int*) 0x8035FDF0;  // @system, (() => cast(int*) 0x8035FDF0) is @system
+int* x3 = getPtr();               // @system, (() => getPtr()) is @system
+int  x4 = getVal();               // @safe, int is not an unsafe type
+@system int x5 = 1;               // @system as requested
+@trusted int* x6 = getPtr();      // @safe, the getPtr call gets trusted
+@safe int* x7 = getPtr();         // error: cannot initialize @safe variable with @system initializer
+
+struct S {
+    // same rules for fields:
+    int* x9 = x3; // @system
+    int  x8 = x5; // @safe
+}
+```
+
+An exception to the last rules is made on unsafe types when the compiler knows the resulting value is safe.
+```D
+int* getNull() pure @system {return null;}
+int* n = getNull(); // despite unsafe type with @system initialization expression, inferred as @safe
+```
+
+Annotations with a scope (`@system {}`) or colon (`@system:`) affect variables just like they do functions.
+```D
+@system {
+    int y0; // @system
+}
+
+@system:
+int y1; // @system
+```
+
+**(4) `__traits(getFunctionAttributes)` may be called on variables and fields**
```

Comment:
That would cause `__traits(getFunctionAttributes)` to be a misnomer.

Reply:
See: https://github.com/dlang/DIPs/pull/179/files/6104d1fcc6fe2eac328cd63d42e15444b817efd5#r341413460
I'm open for suggestions.